tests/includes/class-test-migration.php

```diff
@@ -17,13 +17,103 @@
  */
 class Test_Migration extends ActivityPub_TestCase_Cache_HTTP {
 
+	/**
+	 * Test fixture.
+	 *
+	 * @var array
+	 */
+	public static $fixtures = array();
+
+	/**
+	 * Set up the test.
+	 */
+	public static function set_up_before_class() {
+		\remove_action( 'transition_post_status', array( \Activitypub\Scheduler\Post::class, 'schedule_post_activity' ), 33 );
+		\remove_action( 'transition_comment_status', array( \Activitypub\Scheduler\Comment::class, 'schedule_comment_activity' ), 20 );
+		\remove_action( 'wp_insert_comment', array( \Activitypub\Scheduler\Comment::class, 'schedule_comment_activity_on_insert' ) );
+
+		// Create test posts.
+		self::$fixtures['posts'] = self::factory()->post->create_many( 3, array( 'post_author' => 1 ) );
+
+		$modified_post_id = self::factory()->post->create(
+			array(
+				'post_author'  => 1,
+				'post_content' => 'Test post 2',
+				'post_status'  => 'publish',
+				'post_type'    => 'post',
+				'post_date'    => '2020-01-01 00:00:00',
+			)
+		);
+		self::factory()->post->update_object( $modified_post_id, array( 'post_content' => 'Test post 2 updated' ) );
+
+		self::$fixtures['posts'][] = $modified_post_id;
+		self::$fixtures['posts'][] = self::factory()->post->create(
+			array(
+				'post_author'  => 1,
+				'post_content' => 'Test post 3',
+				'post_status'  => 'publish',
+				'post_type'    => 'page',
+			)
+		);
+		self::$fixtures['posts'][] = self::factory()->post->create(
+			array(
+				'post_author'  => 1,
+				'post_content' => 'Test post 4',
+				'post_status'  => 'publish',
+				'post_type'    => 'post',
+				'meta_input'   => array(
+					'activitypub_content_visibility' => ACTIVITYPUB_CONTENT_VISIBILITY_LOCAL,
+				),
+			)
+		);
+
+		// Create test comment.
+		self::$fixtures['comment'] = self::factory()->comment->create(
+			array(
+				'comment_post_ID'  => self::$fixtures['posts'][0],
+				'user_id'          => 1,
+				'comment_content'  => 'Test comment',
+				'comment_approved' => '1',
+			)
+		);
+	}
+
+	/**
+	 * Tear down the test.
+	 */
+	public static function tear_down_after_class() {
+		// Clean up posts.
+		foreach ( self::$fixtures['posts'] as $post_id ) {
+			\wp_delete_post( $post_id, true );
+		}
+
+		// Clean up comment.
+		if ( isset( self::$fixtures['comment'] ) ) {
+			\wp_delete_comment( self::$fixtures['comment'], true );
+		}
+	}
+
 	/**
 	 * Tear down the test.
 	 */
 	public function tear_down() {
 		\delete_option( 'activitypub_object_type' );
 		\delete_option( 'activitypub_custom_post_content' );
 		\delete_option( 'activitypub_post_content_type' );
+
+		// Clean up outbox items.
+		$outbox_items = \get_posts(
+			array(
+				'post_type'      => 'ap_outbox',
+				'posts_per_page' => -1,
+				'post_status'    => 'draft',
```

Comment:
I think we should use the status to track the process of an item. If an item is added, I would set it to `pending` and if it was sent to the fediverse, I would change that to `publish`.